__tests__/DMT.test.js

```diff
@@ -0,0 +1,221 @@
+                    version: '2.9.2',
+                  },
+                },
+                integrations: {
+                  All: true,
+                },
+                messageId: '1659505271412300-2d882451-7f50-4f23-b5ac-919fa8a1957d',
+                name: 'page view 123',
+                originalTimestamp: '2022-08-03T05:41:11.412Z',
+                properties: {},
+                type: 'page',
+              },
+            },
+          },
+        ],
+      },
+    ],
+  };
+
+  const samplePayloadPartialSuccess = {
+    transformedBatch: [
+      {
+        id: '2CO2YmLozA3SZe6JtmdmMKTrCOl',
+        payload: [
+          {
+            orderNo: 1659505271417,
+            status: '200',
+            event: {
+              message: {
+                anonymousId: '7105960b-0174-4d31-a7a1-561925dedde3',
+                channel: 'web',
+                context: {
+                  library: {
+                    name: 'RudderLabs JavaScript SDK',
+                    version: '2.9.2',
+                  },
+                },
+                integrations: {
+                  All: true,
+                },
+                messageId: '1659505271412300-2d882451-7f50-4f23-b5ac-919fa8a1957d',
+                name: 'page view 123',
+                originalTimestamp: '2022-08-03T05:41:11.412Z',
+                properties: {},
+                type: 'page',
+              },
+            },
+          },
+        ],
+      },
+      {
+        id: '2CO2YmLozA3SZe6JtmdmMKTrCKr',
+        payload: [
+          {
+            orderNo: 1659505271418,
+            status: '400',
+          },
+        ],
+      },
+    ],
+  };
+
+  const xhrMock = {
+    open: jest.fn(),
+    setRequestHeader: jest.fn(),
+    onreadystatechange: jest.fn(),
+    send: jest.fn(),
+    readyState: 4,
+  };
+  const xhrMockSuccess = {
+    ...xhrMock,
+    response: JSON.stringify(samplePayloadSuccess),
+    status: 200,
+  };
+
+  let xhrMockPartialSuccess = {
+    ...xhrMock,
+    attempt: 0,
+    response: JSON.stringify(samplePayloadPartialSuccess),
+    status: 200,
+  };
+
+  xhrMockPartialSuccess['send'] = () => {
+    xhrMockPartialSuccess.onreadystatechange();
+    xhrMockPartialSuccess.attempt++;
+  };
+
+  const xhrMockAccessDenied = {
+    ...xhrMock,
+    response: JSON.stringify({}),
+    status: 404,
+  };
+
+  let xhrMockServerDown = {
+    attempt: 0,
+    open: jest.fn(),
+    setRequestHeader: jest.fn(),
+    onreadystatechange: jest.fn(),
+    readyState: 4,
+    response: null,
+    status: 500,
+  };
+  xhrMockServerDown['send'] = () => {
+    xhrMockServerDown.onreadystatechange();
+    xhrMockServerDown.attempt++;
+  };
+
+  it('Validate payload format', () => {
+    expect(typeof payload).toBe('object');
+    expect(payload.hasOwnProperty('batch')).toBe(true);
+    expect(typeof payload.batch[0]).toBe('object');
+    expect(typeof payload.batch[0].orderNo).toBe('number');
+    expect(payload.batch[0].event).toBe(event);
```

Comment:
createPayload fn has a dynamic part(orderNo is dynamically added). It doesn’t make sense to create another structure to validate response where we will only validate event in response. 
We can do that like `expect(payload.batch[0].event).toBe(event);` or `expect(payload.batch[0].event).toEqual(event);`. This will also validate the event.

Reply:
You can use mocks to solve this issue.

Reply:
Addressed.